src/main/java/com/example/airportproject/controller/FlightController.java

```diff
@@ -1,23 +1,85 @@
 package com.example.airportproject.controller;
 
-import com.example.airportproject.domain.Flight;
+import com.example.airportproject.model.Flight;
+import com.example.airportproject.exception.FlightNotFoundException;
 import com.example.airportproject.repository.FlightRepository;
+import com.example.airportproject.service.FlightService;
+import jakarta.validation.Valid;
+import jakarta.validation.constraints.NotNull;
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.web.bind.annotation.GetMapping;
-import org.springframework.web.bind.annotation.RestController;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.servlet.support.ServletUriComponentsBuilder;
 
+import java.net.URI;
 import java.util.List;
+import java.util.UUID;
 
 @RestController
+@RequestMapping("api/flights")
 public class FlightController {
-    //private FlightService
+    private final FlightService flightService;
+    private final FlightRepository flightRepository;
 
-    private FlightRepository flightRepository;
+    @Autowired
+    public FlightController(FlightRepository flightRepository, FlightService flightService) {
+        this.flightRepository = flightRepository;
+        this.flightService = flightService;
 
-    @GetMapping("/getAll")
-    public List<Flight> getUsers() {
-        return flightRepository.findAll();
     }
 
-    // other controller methods
+    @GetMapping("/")
+    public ResponseEntity<List<Flight>> getFlights() {
+        List<Flight> flights = flightRepository.findAll();
+        return ResponseEntity.ok()
+                .body(flights);
+    }
+
+    @GetMapping("/{id}")
+    public ResponseEntity<Flight> getFlightById(@PathVariable UUID id){
+        Flight found = flightRepository.findById(id).orElseThrow(FlightNotFoundException::new);
+        return ResponseEntity.status(HttpStatus.FOUND)
+                .body(found);
+    }
+
+    @PostMapping("/")
+    public ResponseEntity<Flight> createFlight(@Valid @NotNull @RequestBody Flight flight){
+        Flight created = flightService.createFlight(flight);
+        URI location = ServletUriComponentsBuilder
+                .fromCurrentRequest()
+                .path("/{id}")
+                .buildAndExpand(created.getId()).toUri();
+
+        return ResponseEntity.created(location)
+                .body(created);
+    }
+
+    @PatchMapping("/{id}")
+    public ResponseEntity<Flight> updateStatus(@PathVariable UUID id, @RequestParam(name = "status", required = true) String status){
+            Flight updated = flightService.updateStatus(id, status);
```

Comment:
Look much better! Maybe just rename vars and methods to `patch` just incase maintainers get confused and think you're updating when you're really patching.